### PR TITLE
Use premultiplied alpha blending for everything

### PR DESCRIPTION
**Pull request by mstange**
I've split this up into many small commits for easier reviewing. The only part I'm not entirely sure about is the brush primitive. Wrench reftests pass.

I'm going to push to try in a minute.

r? @glennw

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/2021)

### CODE REVIEW

File: webrender/src/device.rs [hunk 1]
```diff
@@ -1882,16 +1882,6 @@ impl Device {
         self.gl.blend_equation(gl::FUNC_ADD);
     }
 
-    pub fn set_blend_mode_alpha(&self) {
```
Comment (kvark) [inline webrender/src/device.rs]: gonna miss you, old good alpha blending

**Comment by mstange** [comment]
Try push at https://treeherder.mozilla.org/#/jobs?repo=try&revision=1ac30cc080ba829675e21b2e4940a00fec47a062

**Comment by mstange** [comment]
This breaks transparent text-shadow colors, and mix-blend-mode. Looking into it.

**Comment by mstange** [comment]
I add the missing text-shadow color premultiplication, and fixed the composite shader.

r? @kvark

**Comment by mstange** [comment]
I also added two tests.

**Comment by kvark** [comment]
Reftest failure:
> reftests/text/shadow-grey-transparent.yaml == reftests/text/shadow-grey-ref.yaml

**Comment by mstange** [comment]
I've adjusted the fuzz.

**Comment by kvark** [comment]
@mstange thanks! Awesome work, but also notable persistence in eliminating those pesky non-premultiplied colors :) Would you mind launching a final gecko try push before proceeding?

**Comment by mstange** [comment]
Try push is at https://treeherder.mozilla.org/#/jobs?repo=try&revision=63b23c780544973dcb7a588b935e3f4d04b96e65 .

**Comment by mstange** [comment]
The reftests on the try push are green.

**Pull request merged**